### PR TITLE
feat: add support for GitLab CI

### PR DESCRIPTION
**Pull request by joao10lima**
- Unlock GitLab CI as a platform in Pipelinit.
- Add the Python stack for GitLab CI

Pipelinit generates a main `.gitlab-ci.yml` and in this file
is declared the stages for the stack. Inside the folder `.gitlab-ci`
is created the files with the scrips and necessary images.

https://github.com/pipelinit/pipelinit-cli/issues/54

To test:
- `deno test --unstable --allow-all`
- Test on a Python project in GitLab

Examples:
- https://gitlab.com/joao10lima/pipelinit-sample-python
- https://gitlab.com/joao10lima/pipelinit-sample-python/-/pipelines/431335910

image
image
image

### CODE REVIEW

**Review by oesgalha — CHANGES_REQUESTED** [review]
Well done!! 👏 👏 

Can you rebase to fix the README conflict and update the file headers version (pipelinit 0.3.0 instead of 0.2.2)?

I've also included a couple of notes about the generated files, can you take a look?

File: core/templates/gitlab/python/.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  stage: lint
+  image: python:<%= it.version %>
+  needs: []
+  script:
+    - echo "Starting the Lint stage"
+  only:
+    refs:
+      - branches
+    changes:
+      - "**/*.py"
```
[the remark this answers is not in the document]
Comment (joao10lima) [reply core/templates/gitlab/python/.gitlab-ci.yml]: Nice, I managed to test it here:
https://gitlab.com/joao10lima/pipelinit-sample-python/-/pipelines/438684483
image

It works just fine!

File: core/templates/gitlab/python/format.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<% if (it.formatters) { -%>
+format:
+  stage: format
+  image: python:3.6.10
```
Comment (oesgalha) [inline core/templates/gitlab/python/format.yaml:4]: Use the discovered Python version (`it.version`) instead of a fixed one

**Review by oesgalha — CHANGES_REQUESTED** [review]
Great! 😁 

There is one last detail in the Python templates:

**Pull request merged**